src/hotspot/share/utilities/waitBarrier_generic.hpp

```diff
@@ -26,29 +26,68 @@
 #define SHARE_UTILITIES_WAITBARRIER_GENERIC_HPP
 
 #include "memory/allocation.hpp"
+#include "memory/padded.hpp"
 #include "runtime/semaphore.hpp"
 #include "utilities/globalDefinitions.hpp"
 
-// In addition to the barrier tag, it uses two counters to keep the semaphore
-// count correct and not leave any late thread waiting.
 class GenericWaitBarrier : public CHeapObj<mtInternal> {
+private:
+  class Cell : public CHeapObj<mtInternal> {
+  private:
+    DEFINE_PAD_MINUS_SIZE(0, DEFAULT_CACHE_LINE_SIZE, 0);
+
```

Comment:
Just reading the padding, it's unclear why the two pads are where they are.
Can you add a comment about why choose these two locations?

Reply:
Wanted to make sure nothing interferes with cells. But now I realize we actually overpad between the cells (due to both pre-cell and post-cell padding), and underpad for the barrier itself! Fixed in new commit.

Reply:
Ok!